### ISSUE TITLE
Warn developers on upgrading a WebExtension

### ISSUE THREAD

**Issue by andymckay** [mozilla/addons-server#3082]
Broken this out into a seperate issue. The real place I can spot is here:

https://www.dropbox.com/s/d4gfk6xumbq2sok/Screenshot%202016-07-12%2013.15.03.png?dl=0

Probably the X should be a warning triangle and it shouldn't all be red, but you get the idea. I don't think we have an MDN page yet so I'll add that in.

Supports: #3061

**Comment by andymckay** [comment]
I'd like to get this done by the time 48 goes out if possible.

**Comment by andymckay** [comment]
MDN page: https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Porting_a_legacy_Firefox_add-on

**Comment by EnTeQuAk** [comment]
Trying to re-use existing styles we already have, does something like that work? (only the warning icon is new and might need some tweaking)

![screenshot from 2016-07-14 17-55-56](https://cloud.githubusercontent.com/assets/139033/16846336/4cb7cb12-49ec-11e6-8b27-0a71860c2e85.png)

**Comment by andymckay** [comment]
Your style is far better than mine, thank you. Going to ping @jvillalobos for an r? on the text.

**Comment by jvillalobos** [comment]
Suggested: "Thank you for migrating to WebExtensions. Please note that WebExtensions can't be updated back to versions using legacy APIs, so future updates to this add-on will need to also be WebExtensions."

@atsay, what do you think?

**Comment by atsay** [comment]
Apologies for the delay, I didn't get a notification (checking my filters after this).

"Thank you for updating your add-on to WebExtensions. Please note that this is a one-way conversion: once you update to WebExtensions, you cannot submit a version that uses legacy APIs. All future updates to this add-on must also be WebExtensions.

Read more: Porting a legacy Firefox add-on on MDN"

**Comment by andymckay** [comment]
Cool @EnTeQuAk want to finish this off? Thanks all.

**Comment by EnTeQuAk** [comment]
Thanks for the text input. I'll finish this up then, still fighting at how to implement that properly but I'll get it done in time for the next deploy.

**Comment by andymckay** [comment]
Any chance we could get this in for next week @EnTeQuAk please?

**Comment by EnTeQuAk** [comment]
Sorry, I didn't manage to get this in before my PTO and fixed other things for this tag, I'll work on this this week though so that it will land next week if that's alright?

**Comment by eviljeff** [comment]
I'm clearing this milestone - please add one back when there is a commit that is likely to land.

**Comment by muffinresearch** [comment]
@EnTeQuAk did you have something already in progress for this?

**Comment by EnTeQuAk** [comment]
I do have the layout prototype and started patching through the state so that we could show the warning properly bud didn't manage yet to finalize the patch unfortunately and it's still dirty. Will do another go at it asap since I guess it's quite important to have that warning in place.

**Comment by muffinresearch** [comment]
@EnTeQuAk ok let me know if I can help. Could you see if you can fit it in for next week?

**Comment by EnTeQuAk** [comment]
Absolutely, I started cleaning up my branch and I'll try to see that I can put up a pull request tomorrow or latest by Friday. It's mostly tests and cleaning up along the way.

**Comment by ValentinaPC** [comment]
Verified as fixed on AMO-dev FF48.0.2(Win 7)
Postfix screenshot:
![addontowebext](https://cloud.githubusercontent.com/assets/15685960/18626721/60b20de2-7e5f-11e6-96ed-9f59a51ff73d.png)

@EnTeQuAk : should we see the warning red icon too? if yes, can we fix that here or log a separate-small issue?
Thanks!

**Comment by EnTeQuAk** [comment]
> 1) the warning red triangle icon is not displayed

I forgot to upload that, I pushed it in https://github.com/mozilla/addons-server/pull/3532/ and it'll be fixed once it lands.

2) if I update my add-on to a webextension, I receive the message in dev hub as expected, but after installing the latest version (that is a webextension) I am also able to install the previous one (which is an add-on);
add-on used: https://addons-dev.allizom.org/en-US/firefox/addon/add-on-beta/versions/

What do you mean by "install", upload a legacy add-on if your current one is a webextension? If you mean that, it'll be fixed in https://github.com/mozilla/addons-server/pull/3532/ too

**Comment by ValentinaPC** [comment]
My second questions refers to the message displayed when upgrading an add-on to web-estension status: 
![message](https://cloud.githubusercontent.com/assets/15685960/18676069/0ee9ea3a-7f5d-11e6-8ea5-b3b811d2395c.png)

**Comment by EnTeQuAk** [comment]
Hmm, in theory, if you install addon foo-0.1 which is a webextension and try to update to foo-0.2 which is a legacy add-on firefox won't do it. Or, at least we don't support it, I don't know if it'll fail or not.

**Comment by EnTeQuAk** [comment]
https://github.com/mozilla/addons-server/pull/3532 got merged so the icon should be on -dev any minute now :)

**Comment by ValentinaPC** [comment]
Verified as fixed in AMO-dev FF48.0.2(Win 7).
Postfix warning icon:
![warning_icon](https://cloud.githubusercontent.com/assets/15685960/18703619/506e748e-7fef-11e6-859b-2f10d208451d.png)

@EnTeQuAk :
Also, regarding the message posted while upgrading a legacy add-on to a webextesion, maybe we should write instead of "install": "upgrade", or I don't know. Because the upgrade is not supported, but user can install first version of an add-on (that is a legacy add-on), install the last version of that add-on (that is a web-extension) and returning back to legacy add-on by installing its first version.

**Comment by muffinresearch** [comment]
Going to mark this closed again based on https://github.com/mozilla/addons-server/issues/3082#issuecomment-248335236
